src/precice/impl/DataContext.hpp

```diff
@@ -91,25 +91,25 @@ class DataContext {
   DataContext(mesh::PtrData data, mesh::PtrMesh mesh);
 
   /// Defines the mapping associated to this DataContext. A DataContext may also exist without a mapping.
-  MappingContext _mappingContext;
+  std::vector<MappingContext> _mappingContext;
 
-  /// Data this participant will write to and read from
+  /// Unique data this context is associated with
   mesh::PtrData _providedData;
 
   /// If a mapping exists, mesh::PtrData the mapping maps from.
-  mesh::PtrData _fromData;
+  std::vector<mesh::PtrData> _fromData;
 
   /// If a mapping exists, mesh::PtrData the mapping maps from.
-  mesh::PtrData _toData;
+  std::vector<mesh::PtrData> _toData;
 
   /**
-   * @brief Helper to set _mappingContext, _fromData and _toData.
+   * @brief Helper to add _mappingContext, _fromData and _toData.
    *
    * @param mappingContext MappingContext this DataContext will be associated to.
    * @param fromData Data the mapping maps from.
    * @param toData Data the mapping maps to.
    */
-  void setMapping(MappingContext mappingContext, mesh::PtrData fromData, mesh::PtrData toData);
+  void addMapping(MappingContext mappingContext, mesh::PtrData fromData, mesh::PtrData toData);
```

Comment:
I slightly prefer `appendMapping`, because this tells me that there is something listlike which contains mappings and I'm appending my mapping to this thing.